src/main/sensors/current.c

```diff
@@ -107,9 +107,10 @@ static int32_t currentMeterADCToCentiamps(const uint16_t src)
     const currentSensorADCConfig_t *config = currentSensorADCConfig();
 
     int32_t millivolts = ((uint32_t)src * ADCVREF) / 4096;
-    millivolts -= config->offset;
+    // y=x/m+b m is scale in (mV/10A) and b is offset in (mA)
+    float amps = millivolts / (float)config->scale * 10 + (float)config->offset / 1000;
 
-    return (millivolts * 1000) / (int32_t)config->scale; // current in 0.01A steps
+    return amps*100; // Convert back to Centiamps to maintain compatability with the rest of the code
```

Comment:
`return ((millivolts * 10000) / config->scale + config->offset) / 10` should do the same without converting to float first ...